base/applications/mspaint/history.cpp

```diff
@@ -209,6 +209,45 @@ void ImageModel::InvertColors()
     NotifyImageChanged();
 }
 
+void ImageModel::GSColors()
+{
+    int x, y;
+    COLORREF pixel_color;
+
+    CopyPrevious();
+
+    for (x = 0; x < GetWidth(); x++)
+    {
+        for (y = 0; y < GetHeight(); y++)
+        {
+            pixel_color = GetPixel(hDrawingDC, x, y);
+            pixel_color = (pixel_color & 0xFF) * 0x10101;
+            SetPixel(hDrawingDC, x, y, pixel_color);
+        }
+    }
+    NotifyImageChanged();
+}
+
+void ImageModel::BKColors()
```

Comment:
Same as above, basically.
The proposed implementation merely imitates the effects of reducing the bit depth to 1BPP.
Besides, it does so in the slowest way possible, namely via GetPixel and SetPixel.

Reply:
Yes, as before, it is a workaround, better than leave unimplemented function.

Reply:
I'm a bit hesitant, to be honest.
Faking the B&W mode while the image model does not support it properly is somewhat justifiable, but I cannot find a grayscale option anywhere in the Paint versions of Windows 2003 and Windows Vista.
ReactOS currently aims for Windows 2003 compatibility. ReactOS' Paint imitates Vista's version, because it is the most recent one with the traditional GUI.
I believe that limiting this pull request to fake B&W support (as found in the relevant Paint versions) would be the best solution, because it minimizes changes especially to the resource files.
It could otherwise literally take years before all the translations are up to date, again.
Adding a makeshift grayscale implementation now just to replace it later, while the relevant Microsoft Paint versions do not have that functionality in the first place, does not quite sound right.

Reply:
It's not faking B&W and Grayscale. I have implemented two, as you told before, cosmetic effects of image drawed on paint and it works.
Yes, the powerful solution is to change the bit depth from 24BPP to 8, or 1 relating to mode setted. I agree on this. 
So I can leave a comment before the two functions (BSColors and GSColors) to remind future developers that the functions should be improved when there will be an implementation with underlying pixel formats.
For translations, there are many texts not translated and left english for all languages, so this new label certainly does not change everything.